janusgraph-inmemory/src/main/java/org/janusgraph/diskstorage/inmemory/BufferPageUtils.java

```diff
@@ -92,6 +93,30 @@ public static SharedEntryBuffer readFrom(DataInputStream in) throws IOException
         }
     }
 
+    static int computeValPosSize(Entry e)
```

Comment:
I could change it here, but not sure how it could help reuse it in BufferPage - here we have an "unpacked" entry and decide how we are going to pack, there we get the packed entry and need to find out how it WAS packed - so it needs to take the byte array etc...

Reply:
What I noticed is, in BufferPage.java, you always retrieve `valPos` before retrieving `valPosSize`. Since you have `valPos` available, you could call `BufferPageUtils::computeValPosSize(int valPos)` to retrieve `valPosSize`.

The caller in this class should then call `computeValPosSize((array[start+i]).getValuePosition())` to retrieve `valPosSize`.

Reply:
ok, got you now - done